Slightly clarify the build metadata section

matthijskooijman opened the pull request:
Multiple people (#291, #249, #230, #293) have missed the fact that the
build metadata can contain multiple _dot-separated_ identifers, and were
confused enough to report a bug about the dots present in one of the
examples. This attempts to make it more obvious that multiple
identifiers might be used.

This closes #293.

Haacked commented on semver.md:
```diff
@@ -102,8 +102,10 @@ intended compatibility requirements as denoted by its associated
 normal version. Examples: 1.0.0-alpha, 1.0.0-alpha.1, 1.0.0-0.3.7,
 1.0.0-x.7.z.92.
 
-1. Build metadata MAY be denoted by appending a plus sign and a series of dot
```
So my concern here is that the language used to define build identifiers was pretty much exactly the same as the language used above to describe pre-releases. Why is it more confusing for build identifiers? Should we change the description of pre-release too?

matthijskooijman replied:
Excellent point. To be honest, I wasn't interested in pre-releases, so I skipped that section when reading the spec. It makes sense to keep them consistent, so I'll update my PR.

matthijskooijman commented:
Commit added to update the pre-release description as well. If you'd rather have them merged into a single commit, let me know.

jwdonahue commented:
I think this is a non-breaking change that is mildly better than the original.

@Haacked, what's holding up acceptance of this change?

FichteFoll commented on semver.md:
```diff
@@ -102,8 +104,10 @@ intended compatibility requirements as denoted by its associated
-1. Build metadata MAY be denoted by appending a plus sign and a series of dot
-separated identifiers immediately following the patch or pre-release version.
+1. Build metadata MAY be denoted by appending a plus sign followed by one
+or more identifiers. The plus immediately follows the patch or
```
Should also say "plus sign" like in the line above.

matthijskooijman replied:
Indeed, pushed a fix.

matthijskooijman commented:
I rebased this on top of master (there was a minor conflict) and fixed the comment by @FichteFoll (I amended it into the first commit, since I was rebased anyway).

alexandrtovmach commented:
Closing & re-opening to trigger CI

alexandrtovmach commented:
@haacked @indirect what should we do with this PR? It seems a good patch?
